### PR TITLE
build: BSD tweaks

### PR DESCRIPTION
**Pull request by justinmk**
see commit messages, and  https://github.com/neovim/neovim/issues/6771#issuecomment-302921368

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -457,10 +457,12 @@ find_program(GPERF_PRG gperf)
 
 include(InstallHelpers)
 
+if(DEFINED ENV{MANPREFIX})
+  set(CMAKE_INSTALL_MANDIR ${MANPREFIX}/man)
+endif()
```
Comment (justinmk) [inline CMakeLists.txt]: @jamessan Any reason this would be the wrong thing to do on, say, debian?

Comment (jamessan) [inline CMakeLists.txt]: Nope, I don't see that in the environment.

**Comment by jdebp** [comment]
Hmmm.  I quickly tried a distclean then a build and an install.  The manual pages still went into the wrong place.

**Comment by justinmk** [comment]
@jdebp can you try it now?

**Comment by jdebp** [comment]
I've updated master, done another disctlean, build, and install.  No `/usr/local/share/man/man1/nvim.1` any more.  Hooray!  It is in `/usr/local/man/man1/`.  Excellent.

**Pull request merged**